sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -329,8 +341,32 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
         { coeThresholdDate },
       );
     }
-    coeQuery.orderBy("disbursementSchedule.coeStatus");
-    return coeQuery.getMany();
+    if (paginationOptions.searchCriteria) {
+      coeQuery.andWhere(
+        new Brackets((qb) => {
+          qb.where("user.firstName Ilike :searchCriteria", {
+            searchCriteria: `%${paginationOptions.searchCriteria}%`,
+          })
+            .orWhere("user.lastName Ilike :searchCriteria", {
```

Comment:
will the user have the option to search `full name` all together?

Reply:
Search updated to firstname or lastName or fullname. Just extracted the logic from users data table. Thanks.